Libraries/src/Amazon.Lambda.Serialization.SystemTextJson/DefaultLambdaJsonSerializer.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.Serialization.SystemTextJson.Converters;
+
+namespace Amazon.Lambda.Serialization.SystemTextJson
+{
+    /// <summary>
+    /// Custom ILambdaSerializer implementation which uses System.Text.Json
+    /// for serialization.
+    /// 
+    /// <para>
+    /// If the environment variable LAMBDA_NET_SERIALIZER_DEBUG is set to true the JSON coming
+    /// in from Lambda and being sent back to Lambda will be logged.
+    /// </para>
+    /// </summary>    
+    public class DefaultLambdaJsonSerializer : ILambdaSerializer
+    {
+        private const string DEBUG_ENVIRONMENT_VARIABLE_NAME = "LAMBDA_NET_SERIALIZER_DEBUG";
+        private readonly JsonSerializerOptions _options;
+        private readonly bool _debug;
+
+        /// <summary>
+        /// Constructs instance of serializer.
+        /// </summary>        
+        public DefaultLambdaJsonSerializer()
+        {
+            _options = new JsonSerializerOptions()
+            {
+                IgnoreNullValues = true,
+                PropertyNameCaseInsensitive = true,
+                PropertyNamingPolicy = new AwsNamingPolicy()
+            };
+
+            _options.Converters.Add(new DateTimeConverter());
+            _options.Converters.Add(new MemoryStreamConverter());
+            _options.Converters.Add(new ConstantClassConverter());
+            
+            if (string.Equals(Environment.GetEnvironmentVariable(DEBUG_ENVIRONMENT_VARIABLE_NAME), "true", StringComparison.OrdinalIgnoreCase))
+            {
+                this._debug = true;
+            }            
+        }
+
+        /// <summary>
+        /// Constructs instance of serializer with the option to customize the JsonSerializerOptions after the 
+        /// Amazon.Lambda.Serialization.SystemTextJson's default settings have been applied.
+        /// </summary>
+        /// <param name="customizer"></param>
+        public DefaultLambdaJsonSerializer(Action<JsonSerializerOptions> customizer)
```

Comment:
An alternative would be to make `_options` protected so it can be accessed by a derived class.

If I'm not mistaken, the only way to customize the base class is by constructor since it's the AWS Lambda Runtime for .NET Core that instantiates this class, correct?

Reply:
And by making protected, I meant adding a property. For example,
```
protected JsonSerializerOptions SerializerOptions => _options;
```

Reply:
I need the `_options` object so I can add my own `Serialize(object,Type)` method to the serializer (needed by my middleware).

I was hoping I could use the `Action<>` callback in the constructor to store a reference, but alas, that's not legal in C#.
```csharp
// does not work, because the lambda function has no 'this' context in the 'base' invocation
public LambdaJsonSerializer() : base(options => _myOptions = options) { }
```

Reply:
I turned the options into a protected property;